### PR TITLE
Boss talk dialog

### PR DESCRIPTION
**Pull request by JigyasuMittal**
-fix weapons overlapping dialog boxes
-fix single enter button completes and dismisses dialog
-added java docs

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/components/DialogComponent.java [hunk 1]
```diff
@@ -27,6 +28,8 @@ public class DialogComponent extends RenderComponent {
     private static float textLength = 0f;
     private static final float FRAMES_PER_CHAR = 5f;
     private static final float MAX_WIDTH = 15f;
+    
+    private int cooldownTime = 2;
```
Comment (Aryan-S-Gupta) [inline source/core/src/main/com/csse3200/game/components/DialogComponent.java:32]: What is the use of cooldown time ?

File: source/core/src/main/com/csse3200/game/components/DialogComponent.java [hunk 2]
```diff
@@ -78,12 +87,6 @@ public boolean dismissDialog() {
         }
     }
 
-    /**
```
Comment (Aryan-S-Gupta) [inline source/core/src/main/com/csse3200/game/components/DialogComponent.java:81]: Why remove the Javadocs ?

Comment (vsinh14) [reply source/core/src/main/com/csse3200/game/components/DialogComponent.java:81]: He added them

**Review by vsinh14 — APPROVED** [review]
Only added docs LGTM 
functionally identical

File: source/core/src/main/com/csse3200/game/components/DialogComponent.java (hunk 1 above)
Comment (JigyasuMittal) [reply source/core/src/main/com/csse3200/game/components/DialogComponent.java:32]: Earlier there was a bug due to which dialogue box was getting dismissed if player presses enter before dialogue gets displayed completely, so I did this just to resolve that.

**Pull request merged**